### PR TITLE
Add support for async message handling when scheduling storage messag…

### PR DESCRIPTION
**Pull request by geirst**
…es in FileStorManager.

When turned on, the calling thread (e.g. FNET network thread when using Storage API RPC)
gets the next async message to handle (if any) as part of scheduling a storage message.
This async message is then handled by the calling thread immediately,
instead of going via a persistence thread.

@vekterli and @baldersheim please review

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]

File: storage/src/vespa/storage/persistence/asynchandler.h [hunk 1]
```diff
@@ -25,6 +25,7 @@ class AsyncHandler : public Types {
     MessageTrackerUP handlePut(api::PutCommand& cmd, MessageTrackerUP tracker) const;
     MessageTrackerUP handleRemove(api::RemoveCommand& cmd, MessageTrackerUP tracker) const;
     MessageTrackerUP handleUpdate(api::UpdateCommand& cmd, MessageTrackerUP tracker) const;
+    static bool is_async_message(api::MessageType::Id type_id);
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/asynchandler.h:28]: Could safely be tagged as `noexcept`

File: storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp [hunk 2]
```diff
@@ -910,6 +921,24 @@ FileStorHandlerImpl::Stripe::getNextMessage(vespalib::duration timeout)
...
+FileStorHandlerImpl::Stripe::get_next_async_message(monitor_guard& guard)
+{
+    if (_owner.isClosed() || _owner.isPaused()) {
+        return {};
+    }
+    PriorityIdx& idx(bmi::get<1>(*_queue));
+    PriorityIdx::iterator iter(idx.begin()), end(idx.end());
+
+    while ((iter != end) && operationIsInhibited(guard, iter->_bucket, *iter->_command)) {
+        iter++;
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp:934]: Consider `++iter`

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/persistence/asynchandler.h (hunk 1 above)
Comment (geirst) [reply storage/src/vespa/storage/persistence/asynchandler.h]: Fixed

File: storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp (hunk 2 above)
Comment (geirst) [reply storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp]: Fixed

**Pull request merged**